app/components/subject-picker/component.js

```diff
@@ -57,16 +58,44 @@ export default Ember.Component.extend({
     selection2: null,
     selection3: null,
 
-    init() {
-        this._super(...arguments);
+    flatSubjects: Ember.computed('theme.provider.subjectsAcceptable', function() {
+        const acceptableSubjects = this.get('theme.provider.subjectsAcceptable') || [];
+        const flatSubjects = new Set();
+
+        for (const subjects of acceptableSubjects) {
+            for (const subject of subjects[0]) {
+                if (!flatSubjects.has(subject))
+                    flatSubjects.add(subject);
+            }
+        }
+
+        return flatSubjects;
+    }),
+
+    querySubjects(parents = 'null', tier = 0) {
         this.get('store').query('taxonomy', {
             filter: {
-                parents: 'null'
+                parents
             },
             page: {
                 size: 100
             }
-        }).then(results => this.set('_tier1', results));
+        }).then(results => {
+            const flatSubjects = this.get('flatSubjects');
+
+            if (flatSubjects.size) {
```

Comment:
Might not be relevant now that the redirect would keep the user from viewing the preprint under a different provider.